src/Adapter/ArrayAdapter.php

```diff
@@ -21,17 +28,33 @@ public function send($queue, $message)
         return $d['id'];
     }
 
-    public function deleteMessage($receipt, $queue)
+    public function receive($queue)
     {
+        if (!array_key_exists($queue, $this->data)) {
+            throw new OutOfBoundsException("Missing data");
```

Comment:
We could also understand if we want to throw exceptions on missing data or return a null object...

```
return [null, null];
```

Null object allows probably a better data handling 

```
while (list($receipt, $message) = $queue->receive() && $receipt) {
  // use of $message

  $queue->delete($receipt);
}
```